src/components/GlobalSearch/GlobalSearchComponent.tsx

```diff
@@ -0,0 +1,314 @@
+        strings.assets,
+        itemsGroupedByResourceType.asset
+      )}
+      {renderListIfNotNull(
+        searchForFilter.timeSeries,
+        strings.timeSeries,
+        itemsGroupedByResourceType.timeSeries
+      )}
+      {renderListIfNotNull(
+        searchForFilter.events,
+        strings.events,
+        itemsGroupedByResourceType.event
+      )}
+      {renderListIfNotNull(
+        searchForFilter.files,
+        strings.files,
+        itemsGroupedByResourceType.file
+      )}
+    </>
+  );
+}
+
+export function GlobalSearch({
+  strings: userStrings,
+  searchForFilter: userSearchForFilter,
+  onError,
+  renderSearchResult: userRenderSearchResult,
+  renderSearchItem: userRenderSearchItem,
+  onItemSelected: userOnItemSelected,
+  onSearchResults,
+  onItemHover: userOnItemHover,
+}: GlobalSearchProps) {
+  const [searchResults, setSearchResultsState] = useState<SearchItem[] | null>(
+    null
+  );
+  const [loading, setLoading] = useState<boolean>(false);
+  const [searchCounter] = useState<RefCounter>(new RefCounter());
+  const [searchQuery, setSearchQuery] = useState<ApiQuery | null>();
+  const searchForFilter = useMemo(
+    () =>
+      userSearchForFilter == null
+        ? {
+            assets: {},
+            events: {},
+            timeSeries: {},
+            files: {},
+          }
+        : userSearchForFilter,
+    [userSearchForFilter]
+  );
+  const client = useContext(ClientSDKContext);
+
+  const strings = { ...defaultStrings, ...(userStrings || {}) };
+
+  const renderSearchResult =
+    userRenderSearchResult || defaultRenderSearchResult;
+  const renderSearchItem = userRenderSearchItem || defaultRenderSearchItem;
+  const onItemSelected: ItemCallback = item => {
+    if (userOnItemSelected) {
+      userOnItemSelected(item);
+    }
+  };
+  const onItemHover: ItemCallback = item => {
+    if (userOnItemHover) {
+      userOnItemHover(item);
+    }
+  };
+
+  useEffect(() => {
+    if (onSearchResults && searchResults != null) {
+      onSearchResults(searchResults);
+    }
+  }, [searchResults]);
+
+  const handleSearch = (searchQuery: ApiQuery) => {
+    setSearchQuery(searchQuery);
+  };
+  const setSearchResults = (items: SearchItem[] | null) => {
+    setSearchResultsState(items);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    if (searchQuery == null) {
+      return;
+    }
+
+    if (client == null) {
+      throw new Error(
+        'Remember to add a ClientSDKProvider before using any Gearbox components.'
+      );
+    }
+
+    if (searchQuery.query == null || searchQuery.query === '') {
+      setSearchResults(null);
+      return;
+    }
+
+    setLoading(true);
+    // The searchCounter prevents old search results from replacing newer search results
+    const thisId = searchCounter.nextId();
+    // We are not using await here,
+    // because the useEffect function doesn't expect a Promise returned
+    globalSearch(client, searchQuery.query, searchForFilter, 50)
```

Comment:
It's nicer with `.then` and `.catch` IMO in cases like this. `async/await` makes sense when you want promise based code to be written in an imperative manner, but it's not like `then` and `catch` is not useful.

Reply:
sorry for strict words on `.then`, I won't say that `.then` and `.catch` is not useful 🙂. But almost in all components in Gearbox we're trying to use `async/away` approach. `TimeseriesDataExport` has the same thing with using async function inside `useEffect` hook, and it's much easier to read as for me

Reply:
I see, if we have that as a standard for `useEffect` I can do it here. But to use it in `src/components/GlobalSearch/globalSearch` it would become something like `(await someSearch).map(x => withDiscriminator(x, "asset"))` However, if you like `(await ...)` syntax I can go with it.

Reply:
y can move `withDiscriminator()` function inside each search instead, like:
```
const. search = async (filter) => {
  const res = await client.search(filter)

  return withDiscriminator(res, resurceType)
}
```